count spaces

whuan20011 opened the pull request:
no need to find the index of the spaces
tackle with  tail - head - the first max_purchase spaces

whuan20011 committed:
count spaces
no need to find the index of the spaces
tackle with  tail - head - the first max_purchase spaces

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+    fout = open("barn1.out", "w")
+    max_purchase, total_barns, occupied = map(int, fin.readline().strip().split())
+    if max_purchase >= occupied:
+        print >> fout, occupied
```
No space between >> and fout

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+    if max_purchase >= occupied:
+        print >> fout, occupied
+        return
+    arr = []
```
Replace arr with meaningful, like cows

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+    for gate in fin.readlines():
+        arr.append(int(gate.strip()))
+    arr = sorted(arr)
+    brr = []
```
Ditto

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+    arr = sorted(arr)
+    brr = []
+    for i in range(1, len(arr)):
+        brr.append(arr[i] - arr[i - 1])
```
You can replace this loop with [succ - curr for succ, curr in zip(arr[1:], arr)]

zip([1, 2, 3], [2, 3, 4]) => [[1, 2], [2, 3], [3, 4]]
zip([1, 2], [2, 3, 4]) => [[1, 2], [2, 3]]
zip([a2, a3, a4], [a1, a2, a3]) => [[a2, a1],

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+        brr.append(arr[i] - arr[i - 1])
+    brr = sorted(brr)
+    brr = brr[:: -1]
+    count = 0
```
count = sum(brr[:max_purchase - 1])

smy20011 commented on barn2/barn2.py:
```diff
@@ -0,0 +1,28 @@
+    brr = []
+    for i in range(1, len(arr)):
+        brr.append(arr[i] - arr[i - 1])
+    brr = sorted(brr)
```
You can just sort from largest to smallest using sorted(brr, reverse=True)

whuan20011 committed:
no spaces between >> and fout
give meaningful names
use zip to get the spaces
use sorted(list, reverse=True) to make a list from largest to smallest
